docs/quickstarts/astro.mdx

```diff
@@ -0,0 +1,211 @@
+
+</SignedIn>
+
+<SignedOut>
+
+1. Navigate to the Clerk Dashboard.
+2. In the navigation sidebar, select [API Keys](https://dashboard.clerk.com/last-active?path=api-keys).
+3. In the **Quick Copy** section, copy your Clerk publishable and secret key.
+4. Paste your keys into your `.env.local` file.
+
+The final result should resemble the following:
+
+</SignedOut>
+
+  <InjectKeys>
+
+  ```sh filename=".env"
+  PUBLIC_CLERK_PUBLISHABLE_KEY={{pub_key}}
+  CLERK_SECRET_KEY={{secret}}
+  ```
+
+  </InjectKeys>
+
+
+### Add TypeScript declarations
+
+Inside your `src/` directory, update the `.env.d.ts` file to ensure that the locals added by the middleware are properly typed.
+
+```ts filename="src/.env.d.ts" {3}
+/// <reference types="astro/client" />
+/// <reference types="@clerk/astro/env" />
+```
+
+### Update `astro.config.mjs`
+
+To configure Clerk in your astro application, you will need to update your `astro.config.mjs`.
+
+- Add the Clerk integration to the `integrations` list.
+- Install an [SSR adapter](https://docs.astro.build/en/guides/server-side-rendering/#official-adapters). This is required for the integration to work. For this quickstart we chose the [`@astrojs/node`](https://docs.astro.build/en/guides/integrations-guide/node/) adapter.
+- Set `output` to `server`. This is required when deploying to a host supporting SSR.
+
+```ts filename="astro.config.mjs" {2,3,6-8}
+import { defineConfig } from "astro/config";
+import node from "@astrojs/node";
+import clerk from "@clerk/astro";
+
+export default defineConfig({
+  integrations: [clerk()],
+  adapter: node({ mode: "standalone" }),
+  output: "server",
+});
+```
+
+### Add middleware to your application
+
+[`clerkMiddleware()`](/docs/references/astro/clerk-middleware) grants you access to user authentication state throughout your application, on any route or page. It also allows you to protect specific routes from unauthenticated users. To add `clerkMiddleware()` to your application, follow these steps:
+
+1. Create a `middleware.ts` file inside your `src/` directory.
+2. In your `middleware.ts` file, export an `onRequest` constant and assign the result of the `clerkMiddleware` function to it.
+
+```tsx filename="src/middleware.ts"
+import { clerkMiddleware } from "@clerk/astro/server";
+
+export const onRequest = clerkMiddleware();
+```
+
+3. By default, `clerkMiddleware()` will not protect any routes. All routes are public and you must opt-in to protection for routes. See the [`clerkMiddleware()` reference](/docs/references/astro/clerk-middleware) to learn how to require authentication for specific routes.
+
+### Add components to your app
+
+Clerk offers components that allow you to protect your pages. These components are used to control the visibility of your pages based on the user's authentication state.
+
+
+```astro filename="src/layouts/SiteLayout.astro"
```

Comment:
I think for quickguides a simple example is enough to demonstrate how to use components imported from clerk.
```suggestion
```astro filename="src/pages/index.astro"
```

Reply:
I'm going to disagree here. I think this Quickstart, while it doesn't have the same steps as the Next.js one, has the same ultimate approach. A default Next app has a layout and page. Astro doesn't, but its a common pattern and we're implementing it here. 

I would be fine if we trimmed it down `<style>` section but I think would like to keep the `src/pages/index.astro` implementation here.

Reply:
I appreciate your opinion. The intention of this section is to demonstrate the use of pre-built components from the Clerk package in Astro. I think it could go further and include an optional step using React components. In my writing, I try to stay neutral on opinions or structure, focusing instead on showcasing features for users to use as they see fit.